src/transformers/utils/quantization_config.py

```diff
@@ -180,6 +189,117 @@ def update(self, **kwargs):
         return unused_kwargs
 
 
+@dataclass
+class HqqConfig(QuantizationConfigMixin):
+    """
+    Main HqqConfig.
+    Args:
+        nbits (`int`):
+            Number of bits.
+        group_size (`int`):
+            Group-size value.
+        quant_zero (`bool`):
+            Quantize the zero-point.
+        quant_scale (`bool`):
+            Quantize the scaling.
+        offload_meta (`bool`):
+            Offload the meta-data on the CPU.
+        view_as_float (`bool`):
+            View the quantized weight as float (used in distributed training)
+        int (`axis`):
+            Axis along-which grouping is performed.
+        dynamic_config ('dict'):
+            Parameters for dynamic configuration. The key is the name tag of the layer.
+        skip_modules (List[str])::
+            nn,Linear layers to skip.
+        show_progress (bool):
+            Show tqdm quantization for each shard
+        kwargs (`Dict[str, Any]`):
+            Additional parameters from which to initialize the configuration object.
+    """
+
+    def __init__(
+        self,
+        nbits: int = 4,
+        group_size: int = 64,
+        quant_zero: bool = True,
+        quant_scale: bool = False,
+        offload_meta: bool = False,
+        view_as_float: bool = False,
+        axis: int = 0,
+        dynamic_config: dict | None = None,
```

Comment:
```suggestion
        dynamic_config: Optional[dict] = None,
```
This should make the CI happy